### PR TITLE
Sign request only with the body content + resolve userID for API key based session

### PR DESCRIPTION
**Pull request by stijnkoopal**
Signing of full API request is deprecated, only the body should be used: https://beta.doc.bunq.com/basics/authentication/signing

### CODE REVIEW

File: bunq/bunq_test.go [hunk 1]
```diff
@@ -200,12 +200,9 @@ func formatFilePathByName(fileName string) string {
 func sendResponseWithSignature(t *testing.T, w http.ResponseWriter, resCode int, body interface{}) {
 	w.Header().Set("X-Bunq-Client-Response-Id", uuid.NewV4().String())
 	w.Header().Set("Cache-controll", fmt.Sprintf("max-age=%s", uuid.NewV4().String()))
```
Comment (OGKevin) [inline bunq/bunq_test.go]: AFAIK these headers can also be removed as they were there purely for the signature. I think its better to remove unneeded code to remove confusion of why its there.

**Review by OGKevin — CHANGES_REQUESTED** [review]
Thanks for this 🙏🏾!

Just a small code quality comment.

File: bunq/bunq_test.go (hunk 1 above)
Comment (stijnkoopal) [reply bunq/bunq_test.go]: Check, fixed!

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/OGKevin/go-bunq
 
-go 1.12
```
Comment (OGKevin) [inline go.mod:3]: Sorry that I missed this. You forgot to bump the version in https://github.com/OGKevin/go-bunq/blob/500f3254737cdbb9d6d170e39b1609d00096877c/.travis.yml#L3 Do you mind updating it there as well to ensure that the pipeline is running tests on the correct version.

**Review by OGKevin — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (stijnkoopal) [reply go.mod:3]: Done!

**Review by OGKevin — APPROVED** [review]

**Comment by OGKevin** [comment]
thanks @stijnkoopal 🎊

**Pull request merged**